src/features/post/postSlice.js

```diff
@@ -0,0 +1,125 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import {
+  arrayUnion,
+  doc,
+  updateDoc,
+  getDocs,
+  collection,
+} from 'firebase/firestore';
+import { nanoid } from '@reduxjs/toolkit';
+import { db } from '../../firebase';
+import { sortPosts } from '../../utils/utils';
+
+const initialState = {
+  homePosts: [],
+  explorePosts: [],
+  bookmarks: [],
+  newPost: {},
+  homeStatus: 'idle',
+  exploreStatus: 'idle',
+  bookmarkStatus: 'idle',
+  postStatus: 'idle',
+};
+
+export const addPost = createAsyncThunk(
+  'post/addPost',
+  async ({ uid, postText, name, username, photoURL }) => {
+    try {
+      const postRef = doc(collection(db, 'posts'), uid);
+      const postId = nanoid();
+      const newPost = {
+        uid,
+        id: postId,
+        name,
+        username,
+        photoURL,
+        postText,
+        likes: 0,
+        comments: [],
+        uploadDate: new Date().toString(),
+      };
+      await updateDoc(postRef, {
+        posts: arrayUnion({ ...newPost }),
+      });
+      return newPost;
+    } catch (err) {
+      console.error('add post', err);
+    }
+  }
+);
+
+const getAllPost = async () => {
+  try {
+    const querySnapshot = await getDocs(collection(db, 'posts'));
+    let allPosts = [];
+    querySnapshot.forEach(
+      doc => (allPosts = [...allPosts, ...doc.data().posts])
+    );
+    return allPosts;
+  } catch (err) {
+    console.error('get all posts', err);
+  }
+};
+
+export const getHomePosts = createAsyncThunk(
+  'post/getHomePost',
+  async ({ uid, following }) => {
+    try {
+      const allPosts = await getAllPost();
+      const homePosts = allPosts.filter(post =>
+        following.some(item => item.uid === post.uid)
+      );
+      homePosts.push(...allPosts.filter(post => post.uid === uid));
+      return sortPosts(homePosts, 'newest');
+    } catch (err) {
+      console.error('get home posts', err);
+    }
+  }
+);
+
+export const getExplorePosts = createAsyncThunk(
+  'post/getExplorePosts',
+  async ({ uid, following }) => {
+    try {
+      const allPosts = await getAllPost();
+      const filterPosts = allPosts.filter(
+        post => !following.some(item => item.uid === post.uid)
+      );
+      const explorePosts = filterPosts.filter(post => post.uid !== uid);
+      return sortPosts(explorePosts, 'newest');
+    } catch (err) {
```

Comment:
instead of console log you can use 
`return thunkAPI.rejectWithValue('something went wrong');`
so that you can show in [getExplorePosts.rejected].
.
.
Do this for all promises.